packages/babel-traverse/src/scope/index.ts

```diff
@@ -242,6 +242,15 @@ function resetScope(scope: Scope) {
   scope.globals = Object.create(null);
 }
 
+function isAnonymousFunctionExpression(
+  path: NodePath,
+): path is NodePath<t.FunctionExpression | t.ArrowFunctionExpression> {
```

Comment:
The predicate here also converts `NodePath<A> | NodePath<B>` to `NodePath<A | B>` such that `path.pushContainer` can work. This is a bit unfortunate as we have already made it distributive.